### PR TITLE
FSE plugin: add newspack posts block to version control

### PR DESCRIPTION
**Pull request by simison**
Background pbAok1-dw-p2

#### Changes proposed in this Pull Request

* Adds Newspack plugin to FSE `package.json`
* Builds the Newspack posts block via `prepare` script
* Newspack plugin has its own `package.json` with `dependencies` array — which is fantastic because now we get dependencies pulled into our package-lock. I don't think we were listing those dependencies anywhere even if we had the code checked in, which seems fragile.

##### TODO
- [x] Do something about the asset version — maybe we can just use the FSE version here? Alternatively, pull Newspack plugin version from package.json. _We are updating the version from the package.json._
- [x] Someone could be pinged when new plugin release is available by adding `reviewers` or `assignees` to `renovate.json` for this dependency.

#### Testing instructions

- Run `npm distclean`, confirm that `full-site-editing-plugin/blog-posts-block/newspack-homepage-articles` is gone.
- Run `npm i` (at the Calypso root) and confirm that you get freshly built block in the above directory
- Confirm that the block works and is renamed appropriately
- Confirm that updating the version of Newspack in the package.json and then running `npm i` also updates the version of Newspack locally -- includes changing the version defined in full-site-editing-plugin/blog-posts-block/index.php.

### CODE REVIEW

**Comment by noahtallen** [comment]
Nice stuff here!

Another way to test just the script part without all the npm i stuff:
`npx lerna run prepare --scope='@automattic/full-site-editing' --stream --no-prefix`

> Do something about the asset version

I'll try to add this to the prepare script. Should be easy to do!

> Someone could be pinged when new plugin release is available

Using renovate would definitely be ideal here. I'm not sure if it can handle GitHub releases, though, I couldn't find that in the documentation... However, if this package was added to npm, that would be even better!

File: apps/full-site-editing/bin/sync-blog-posts-block.sh [hunk 1]
```diff
@@ -3,43 +3,49 @@
 # try whether user passed --release
 if [ -n "$npm_config_release" ]
 then
-    MODE=release
-    NAME=$npm_config_release
-    URL=https://github.com/Automattic/newspack-blocks/releases/download/$NAME/newspack-blocks.zip
+	MODE=release
```
Comment (noahtallen) [inline apps/full-site-editing/bin/sync-blog-posts-block.sh]: I changed the file from spaces to tabs here

**Comment by noahtallen** [comment]
I did some experimenting, but some of the regex I was trying does not work on macOS, since it does not support PCRE (perl regex). On my sandbox, I'd been using `grep -Po` to run perl regex which would have let us do something like:

```sh
if [ "$MODE" = "npm" ] ; then
	NEW_VERSION=`grep -Eo '"newspack-blocks": "github:Automattic/newspack-blocks#\K[^"]*"' ../package.json`
	sed -i "s~define( 'NEWSPACK_BLOCKS__VERSION', [^']*'~define( 'NEWSPACK_BLOCKS__VERSION', \"${NEW_VERSION}~" $ENTRY
fi
```

If something like renovate runs on a linux box, this would probably work fine, but it wouldn't work locally. Otherwise, I'm not sure how to handle this (perhaps some node scripting)

File: apps/full-site-editing/bin/sync-blog-posts-block.sh (hunk 1 above)
Comment (simison) [reply apps/full-site-editing/bin/sync-blog-posts-block.sh]: Oh, odd that Prettier didn't do its job?

Comment (jsnajdr) [reply apps/full-site-editing/bin/sync-blog-posts-block.sh]: Prettier doesn't format `*.sh` files 🙂

Comment (simison) [reply apps/full-site-editing/bin/sync-blog-posts-block.sh]: <h1>🤦‍♂ </h1>

**Comment by obenland** [comment]
Should I hold off on https://github.com/Automattic/wp-calypso/pull/39270 in favor of this PR?

**Comment by noahtallen** [comment]
@obenland, I'm not sure -- I don't know if we'll have anything shippable by the end of the week, so if that PR is gonna make it in pretty soon, it should be fine to land :)

**Comment by noahtallen** [comment]
@Automattic/ajax do you think that we'll have any issues if we stop including the source code in git? This still keeps the release manual, since we'll have to bump the npm version when we want to release an update.

File: package-lock.json [hunk 2]
```diff
@@ -11501,28 +11501,28 @@
 					"dependencies": {
 						"abbrev": {
 							"version": "1.1.1",
-							"resolved": false,
+							"resolved": "https://registry.npmjs.org/abbrev/-/abbrev-1.1.1.tgz",
```
Comment (noahtallen) [inline package-lock.json]: @Automattic/team-calypso is there anything to be worried about with all these package-lock changes? Don't want to mess anything else up when merging!

File: apps/full-site-editing/package.json [hunk 3]
```diff
@@ -76,6 +77,7 @@
 		"@wordpress/url": "*",
 		"classnames": "2.2.6",
 		"lodash": "*",
-		"moment": "*"
+		"moment": "*",
+		"newspack-blocks": "github:Automattic/newspack-blocks#1.0.0-alpha.26"
```
Comment (noahtallen) [inline apps/full-site-editing/package.json]: Note that we will handle updating to the latest version of the Newspack block in a separate PR.

**Comment by noahtallen** [comment]
Alright, this should be ready to go. I tried to add something for renovate, but I'm honestly not sure if it will work since it's using a GitHub tag to reference the newspack version. (my guess is that it will only work if it's published to the registry.)

The new process would look like:
1. Update the version of the block in full-site-editing/package.json
2. Run `npm i`
3. Test the block in the FSE build.

File: renovate.json [hunk 4]
```diff
@@ -48,6 +48,12 @@
 		{
 			"extends": "monorepo:storybook",
 			"prPriority": 1
+		},
+		{
+			"paths": [ "apps/full-site-editing/package.json" ],
+			"packageName": "newspack-blocks",
+			"reviewers": [ "team:@Automattic/create" ],
```
Comment (simison) [inline renovate.json]: I think we should add Grow focus here since they'll handle everything blocks-related. @obenland @creativecoder

**Review by vindl — APPROVED** [review]
Tested fine for me. :shipit:

**Review by Addison-Stavlo — COMMENTED** [review]
✅  Run npm i (at the Calypso root) and confirm that you get freshly built block in the above directory

❓  Confirm that the block works and is renamed appropriately
 
I am a bit unsure how to do this.  When I build FSE the above directory gets wiped.

✅  Confirm that updating the version of Newspack in the package.json and then running npm i also updates the version of Newspack locally -- includes changing the version defined in full-site-editing-plugin/blog-posts-block/index.php.

File: renovate.json (hunk 4 above)
Comment (noahtallen) [reply renovate.json]: done. since grow isn't a team, i added Ajax

**Comment by noahtallen** [comment]
> When I build FSE the above directory gets wiped.

Thanks for reporting! This should be fixed now.

**Comment by noahtallen** [comment]
Verifying that the blog posts block works after running npm i and then the FSE build:

Screen Shot 2020-03-12 at 7 11 08 PM

**Review by Addison-Stavlo — APPROVED** [review]
Awesome! 
The FSE build doesn't clear it anymore, and Im confirming that the new version runs well in the editor as well.

🎉

**Pull request merged**